packages/components/src/toggle-group-control/toggle-group-control/toggle-group-control-backdrop.tsx

```diff
@@ -1,86 +1,96 @@
-	const [ canAnimate, setCanAnimate ] = useState( false );
-	const [ renderBackdrop, setRenderBackdrop ] = useState( false );
-
-	useEffect( () => {
-		const containerNode = containerRef?.current;
-		if ( ! containerNode ) return;
+	const shouldReduceMotion = useReducedMotion();
+	const { items, state, isBlock } = useToggleGroupControlContext();
 
-		/**
-		 * Workaround for Reakit
-		 */
-		const targetNode = containerNode.querySelector(
-			`[data-value="${ state }"]`
-		);
-		setRenderBackdrop( !! targetNode );
-		if ( ! targetNode ) {
-			return;
-		}
+	const [ itemSizes, setItemSizes ] = useState<
+		| null
+		| {
+				width: number;
+				left: number;
+		  }[]
+	>( null );
 
-		const computeDimensions = () => {
-			const {
-				width: offsetWidth,
-				x,
-			} = targetNode.getBoundingClientRect();
-
-			const { x: parentX } = containerNode.getBoundingClientRect();
-
-			const borderWidth = 1;
-			const offsetLeft = x - parentX - borderWidth;
+	const selectedItemIndex = items.findIndex(
+		// All valid children (e.g. extending `ToggleGroupControlOptionBase`)
+		// have a `data-value` attribute.
+		( item ) =>
+			typeof state !== 'undefined' &&
+			item.ref.current?.dataset.value === `${ state }`
+	);
 
-			setLeft( offsetLeft );
-			setWidth( offsetWidth );
-		};
-		// Fix to make the component appear as expected inside popovers.
-		// If the targetNode width is 0 it means the element was not yet rendered we should allow
-		// some time for the render to happen.
-		// requestAnimationFrame instead of setTimeout with a small time does not seems to work.
-		const dimensionsRequestId = window.setTimeout( computeDimensions, 100 );
+	// `useLayoutEffect` is necessary because we need to wait for the DOM
+	// mutations to take effect before reading the new element's sizes and offsets.
+	useLayoutEffect( () => {
+		setItemSizes(
+			items.map( ( item ) => {
+				// Using `offsetLeft` in order to get the gap with respect to the
+				// `offsetParent` element. This is different from the behavior of other
+				// functions (like `getBoundingClientRect`), which compute those figures
+				// relatively to the viewport.
+				// For this reason, it's important that the `offsetParent` of each `<Radio>`
+				// option is the inner wrapper of `ToggleGroupControl` — which means that in
+				// the DOM tree between that wrapper and each `<Radio>` component there
+				// can't be any element with `position: relative` or `position: absolute`.
+				// See https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetLeft
 
-		let animationRequestId: number;
-		if ( ! canAnimate ) {
-			animationRequestId = window.requestAnimationFrame( () => {
-				setCanAnimate( true );
-			} );
-		}
-		return () => {
-			window.clearTimeout( dimensionsRequestId );
-			window.cancelAnimationFrame( animationRequestId );
-		};
-	}, [ canAnimate, containerRef, containerWidth, state, isAdaptiveWidth ] );
+				const toggleGroupOptionWrapper = item.ref.current?.closest(
```

Comment:
Since the markup of `ToggleGroupControlOptionBase` can change [if the tooltip is enabled or not](https://github.com/WordPress/gutenberg/blob/5b35278128db9d05cf3bc30aa6e50e8b1e640e4a/packages/components/src/toggle-group-control/toggle-group-control-option-base/component.tsx#L32-L41), I decided to add a data attribute to the option's wrapper and use [`Element.closest()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) to find it each item's wrapper.